Lib/test/test_support.py

```diff
@@ -24,6 +26,73 @@
 TESTFN = os_helper.TESTFN
 
 
+class LogCaptureHandler(logging.StreamHandler):
+    """
+    A logging handler that stores log records and the log text.
+
+    Derived from pytest caplog.
+
+    The MIT License (MIT)
+
+    Copyright (c) 2004 Holger Krekel and others
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy of
+    this software and associated documentation files (the "Software"), to deal in
+    the Software without restriction, including without limitation the rights to
+    use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+    of the Software, and to permit persons to whom the Software is furnished to do
+    so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+    """
```

Comment:
About the licence, in general this seems fine to use/modify the code and credit it, but maybe it should be mentioned in https://docs.python.org/3/license.html instead? Let's ask if @encukou knows.

Reply:
That is where we put the licences (see e.g. https://docs.python.org/3/license.html#test-epoll)
But I don't know much about incorporating code where not all authors signed the CLA.

It seems `caplog` is [derived](https://github.com/pytest-dev/pytest/commit/8eafbd05ca2d980b36541fbc9d547e52b6016a9a) from `pytest-catchlog` which has a [different copyright section](https://github.com/eisensheng/pytest-catchlog/blob/develop/LICENSE.txt).

Reply:
how about if I just pare it down to the bare minimum for the tests to pass:

```python
class LogCaptureHandler(logging.StreamHandler):
    def __init__(self):
        super().__init__(io.StringIO())
        self.records = []

    def emit(self, record) -> None:
        self.records.append(record)
        super().emit(record)

    def handleError(self, record):
        if logging.raiseExceptions:
            raise
```

Reply:
is it small enough to not be a concern for copyright?

Reply:
I am not an expert in licensing, but this code sample does not seem like one that requires a license note. A comment would probably be just enough.